lib/ansible/galaxy/collection.py

```diff
@@ -310,6 +310,14 @@ def from_name(collection, apis, requirement, force, parent=None):
 
         for api in apis:
             collection_url_paths = [api.api_server, 'api', 'v2', 'collections', namespace, name, 'versions']
+
+            available_api_versions = get_available_api_versions(api)
```

Comment:
This is can be a pretty expensive operation, especially when we have multiple collections that we are trying to install. We should be setting the available versions once on the `api` object so that once we only need to do this check once for every server.